causallearn/graph/GeneralGraph.py

```diff
@@ -865,20 +865,21 @@ def remove_nodes(self, nodes: List[Node]):
     # nodes of this graph together with the edges between them.
     def subgraph(self, nodes: List[Node]):
         subgraph = GeneralGraph(nodes)
-
+    
         graph = self.graph
-
+    
+        nodes_to_delete = []
+    
         for i in range(self.num_vars):
             if not (self.nodes[i] in nodes):
-                graph = np.delete(graph, i, axis=0)
-
-        for i in range(self.num_vars):
-            if not (self.nodes[i] in nodes):
-                graph = np.delete(graph, i, axis=1)
-
+                nodes_to_delete .append(i)
```

Comment:
Minor: perhaps we could move the space here? 'nodes_to_delete.append(i)' -> nodes_to_delete .append(i)